File: DynamicPlanning/MaxProfit.js
/**
 * 最大收益
 * source: https://leetcode-cn.com/explore/interview/card/top-interview-questions-easy/23/dynamic-programming/55/
 */
 
var maxProfit = function(prices) {
	if (!prices || prices.length == 0 || prices.length == 1) return 0;

	var profit = 0;
	var minValue = prices[0];
	for (var i = 1; i < prices.length; i++) {
		profit = Math.max(profit, prices[i] - minValue);
		minValue = Math.min(minValue, prices[i]);
	}

	return profit;
};

var values = [1,7,8,9,5,3,9];
console.log(maxProfit(values));

File: 初级算法/Swift/Array/maxProfit.swift
/**
 * 买卖股票的最佳时机 II
 * source: https://leetcode-cn.com/explore/interview/card/top-interview-questions-easy/1/array/22/
 */
import Foundation

class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        guard prices.count > 1 else {
            return 0
        }
        
        var maxProfit = 0
        
        for i in 1..<prices.count {
            maxProfit += max(0, prices[i]-prices[i-1])
        }
        
        return maxProfit
    }
}

let prices = [7,6,4,3,1]
print(Solution().maxProfit(prices))


